src/bindings/python/flux/future.py

```diff
@@ -185,6 +187,55 @@ def get(self):
     def incref(self):
         self.pimpl.flux_future_incref()
 
+    # asyncio functions
+
+    def __await__(self):
+        """
+        Await to work with asyncio.
+        """
+        if not hasattr(self, "_loop"):  # pylint: disable=attribute-defined-outside-init
+            import flux.asyncio  # pylint: disable=import-outside-toplevel
+            self._loop = flux.asyncio.loop  # pylint: disable=attribute-defined-outside-init
+
+        if not self.is_ready():
+            self._asyncio_future_blocking = True  # pylint: disable=attribute-defined-outside-init
+
+            # Yielding here tells the task to wait for completion.
+            yield self
+
+        if not self.is_ready():
+            raise RuntimeError("await wasn't used with future")
+        return self.get()
+
+    def add_done_callback(self, func, *, context=None):
+        """
+        Add a callback to be run when the future becomes done.
+        """
+        if self.is_ready():
+            self._loop.call_soon(func, self, context=context)
+        else:
+            def call_wrapper(_, future):
+                # A wrapper to a future to stop the reactor and run the callback
+                func(future)
+                handle = self.get_flux()
+                reactor = handle.get_reactor()
+                handle.reactor_stop(reactor)
+
+            # self.then alone (if passed directly) provides an extra argument
+            # that the asyncio.Task cannot accept
+            self.then(call_wrapper, self)
+
+    def result(self):
+        """
+        Result is called by asyncio Task (in C) to determine done-ness.
+        """
+        if self.is_ready():
+            return self.get()
+        import asyncio.exceptions  # pylint: disable=import-outside-toplevel
```

Comment:
This import should go at the top level.

Reply:
I put it here because I was worried if someone is using an older version of Python this would break their flux python bindings. Is this not the case / impossible to happen?

Reply:
Oh, I didn't realize it was a version thing. We support 3.6+ right now. It's better not to have version-specific code and it doesn't look like the exception you want exists in 3.6 (it comes in 3.7?) so I would just throw a generic `RuntimeError` or something.

Reply:
This is moved to the top - non Python 3 users beware!

Reply:
Actually I think this will break 3.7 too. Like I said above I would actually remove this import entirely and instead just throw a generic RuntimeError or something.

Reply:
oh gotcha], didn't see that. Will do.

Reply:
> non Python 3 users beware!

FYI - Flux requires Python >= 3.6